src/devices/openxrheadset/OpenXrInterface.cpp

```diff
@@ -1038,6 +1087,84 @@ bool OpenXrInterface::updateInteractionProfiles()
     return true;
 }
 
+void OpenXrInterface::updateHandTracking()
+{
+    // allocate memory for left_hand_joint_locations
+    m_pimpl->right_hand_joint_locations.resize(XR_HAND_JOINT_COUNT_EXT);
+
+    // Allocate memory for left_hand_joint_locations (if not already done)
+    m_pimpl->left_hand_joint_locations.resize(XR_HAND_JOINT_COUNT_EXT);
+
+    m_pimpl->leftHandJointPoses_.clear();
+    m_pimpl->rightHandJointPoses_.clear();
```

Comment:
Rather than clearing it, I would suggest "invalidating" the poses, and keeping the size of these vectors constant